server/provider/networkProvider.go

```diff
@@ -45,19 +46,20 @@ type ArgsNewNetworkProvider struct {
 	PubKeyConverter       core.PubkeyConverter
 }
 
+// In the future, we might rename this to "networkFacade" (which, in turn, depends on networkProvider, currencyProvider, blocksProvider and so on).
```

Comment:
Currently, blocks-related logic resides in `provider/networkProvider.go`, while accounts-related logic resides in `provider/accounts.go` (design inconsistency).